sdk/graphql-transport/src/mappers/transaction-block.ts

```diff
@@ -353,3 +343,257 @@ function mapTransactionArgument(arg: typeof bcs.Argument.$inferType): SuiArgumen
 
 	throw new Error(`Unknown argument type ${arg}`);
 }
+
+const OBJECT_DIGEST_DELETED = toB58(Uint8Array.from({ length: 32 }, () => 99));
+const OBJECT_DIGEST_WRAPPED = toB58(Uint8Array.from({ length: 32 }, () => 88));
+const OBJECT_DIGEST_ZERO = toB58(Uint8Array.from({ length: 32 }, () => 0));
+const ADDRESS_ZERO = normalizeSuiAddress('0x0');
+
+export function mapEffects(data: string): SuiTransactionBlockResponse['effects'] {
+	const effects = bcs.TransactionEffects.parse(fromB64(data));
+
+	let effectsV1 = effects.V1;
```

Comment:
Mapping all this back to the JSON RPC representation is a bit involved.  We need to account for both V1 and V2 effects in bcs, and JSON RPC uses a separate effects v1 format that deviates a bit from bcs effects v1.  This function first creates effectsV1 either from bcs effects v1, or by converting v2 down to v1.  Then it has a separate mapping of the bcs v1 format to the JSON rpc version